splinepy/_spline.py

```diff
@@ -338,8 +342,12 @@ def unique_knots(self,):
         """
         logging.debug("Spline - Computing unique knots using `np.unique`.")
         unique_knots = []
-        for k in self.knot_vectors:
-            unique_knots.append(np.unique(k).tolist())
+        if self.whatami.startswith("Bezier"):
+            unique_knots=[[0,1]]*self.para_dim
```

Comment:
Yes and No. It is a convention because it is convenient for compositions, but you could also define a knot vector between two random (increasing) values. Technically, Beziers don't require knot vectors because they are based on Bernstein polynomials, but if you want to convert them into B-splines, you need to define a parametric space.

Reply:
modified in 3a2b33f